src/app/modules/library/select-all-projects-checkbox/select-all-projects-checkbox.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { MAT_CHECKBOX_DEFAULT_OPTIONS } from '@angular/material/checkbox';
+
+@Component({
+  selector: 'select-all-projects-checkbox',
+  templateUrl: './select-all-projects-checkbox.component.html',
+  styleUrls: ['./select-all-projects-checkbox.component.scss'],
+  providers: [{ provide: MAT_CHECKBOX_DEFAULT_OPTIONS, useValue: { clickAction: 'noop' } }]
+})
+export class SelectAllProjectsCheckboxComponent {
+  @Input() selectedAllProjects: boolean = false;
+  @Input() selectedSomeProjects: boolean = false;
+  @Output()
+  selectAllProjectsChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
Put on one line?
Also, should this field be named ```selectAllProjectsEvent```, following Angular convention (https://angular.io/guide/inputs-outputs#configuring-the-child-component-1)?